database/pebble/db.go

```diff
@@ -0,0 +1,341 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package pebble
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"runtime"
+	"sync"
+
+	"github.com/cockroachdb/pebble"
+	"github.com/cockroachdb/pebble/bloom"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"golang.org/x/exp/slices"
+
+	"go.uber.org/zap"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/avalanchego/utils/set"
+	"github.com/ava-labs/avalanchego/utils/units"
+)
+
+const (
+	// pebbleByteOverHead is the number of bytes of constant overhead that
+	// should be added to a batch size per operation.
+	pebbleByteOverHead = 8
+	numLevels          = 7
+
+	targetFileSize = 2 * units.MiB
+	filterPolicy   = bloom.FilterPolicy(10)
+)
+
+var (
+	_ database.Database = (*Database)(nil)
+
+	ErrInvalidOperation = errors.New("invalid operation")
+
+	defaultCacheSize = 512 * units.MiB
+	DefaultConfig    = Config{
+		CacheSize:                   defaultCacheSize,
+		BytesPerSync:                512 * units.KiB,
+		WALBytesPerSync:             0, // Default to no background syncing.
+		MemTableStopWritesThreshold: 8,
+		MemTableSize:                defaultCacheSize / 4,
+		MaxOpenFiles:                4 * units.KiB,
```

Comment:
This should be the number of files, not a size:

https://github.com/cockroachdb/pebble/blob/1efa535da00668c2b50ac1a3497a2623338db01e/options.go#L786-L790

Reply:
This is equivalent to `4_096`, which is intended, but I'll change it to be declared that way to make it more clear